File: Day46/CenterOfStarGraph.java
import java.util.*;

public class CenterOfStarGraph {
    public int findCenter(int[][] edges) {
        if (edges[0][0] == edges[1][0] || edges[0][0] == edges[1][1]) {
            return edges[0][0];
        } else {
            return edges[0][1];
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.println("--- 1791. Find Center of Star Graph (Java) ---");
        System.out.print("Enter the number of edges (N-1): ");
        int numEdges = scanner.nextInt();
        
        int[][] edges = new int[numEdges][2];
        System.out.println("Enter the edges, one pair per line (e.g., 1 2):");

        for (int i = 0; i < numEdges; i++) {
            System.out.print("Edge " + (i + 1) + ": ");
            edges[i][0] = scanner.nextInt();
            edges[i][1] = scanner.nextInt();
        }

        CenterOfStarGraph solution = new CenterOfStarGraph();
        int center = solution.findCenter(edges);

        System.out.println("\nInput Edges: " + Arrays.deepToString(edges));
        System.out.println("Output (Center Node): " + center);

        scanner.close();
    }
}

File: Day46/CenterOfStarGraph.py
class Solution:
    def findCenter(self, edges: list[list[int]]) -> int:
        if edges[0][0] == edges[1][0] or edges[0][0] == edges[1][1]:
            return edges[0][0]
        else:
            return edges[0][1]

def main():
    print("--- 1791. Find Center of Star Graph (Python) ---")
    
    try:
        num_edges = int(input("Enter the number of edges (N-1): "))
    except ValueError:
        print("Invalid number of edges.")
        return

    edges = []
    print("Enter the edges, one pair per line (e.g., 1 2):")

    for i in range(num_edges):
        try:
            edge_input = input(f"Edge {i + 1}: ").split()
            u = int(edge_input[0])
            v = int(edge_input[1])
            edges.append([u, v])
        except (IndexError, ValueError):
            print("Invalid edge input. Please enter two space-separated integers.")
            return

    solution = Solution()
    center = solution.findCenter(edges)

    print(f"\nInput Edges: {edges}")
    print(f"Output (Center Node): {center}")

if __name__ == "__main__":
    main()